### PR TITLE
Update accordion panels styling and functionality

### LINKED ISSUE #45: Review Letter: Panel Transitioning Commands
### Panel Transitioning commands
To ease the transitions across the 3 steps before sending letter, we want to use the 'enter command' to signify moving to the next step and implement locking for the accordion steps. 
Screen Shot 2022-02-10 at 9 36 54 AM

### Tasks
- [x] Create "next" buttons for each accordion panel 
- [ ] Create default, in progress and complete styles for the accordion as seen in the screenshot below:
Screen Shot 2022-02-10 at 9 40 59 AM

- [ ] Implement functionality so user is required to complete the steps in order

### PR DESCRIPTION
**Pull request by tdo95**
For the panel transitioning commands issue #45 the remaining tasks were to:

- Create default, in progress and complete styles for the accordion
- Implement functionality so user is required to complete the steps in order

## Solution

- I updated the default styling of each expansion panel's title, sub title, and panel number using the [color class system](https://vuetifyjs.com/en/styles/colors/) and [typography helper classes](https://vuetifyjs.com/en/styles/text-and-typography/) within Vuetify. The text size, bolding, and color of each element has been changed to match the [design reference photo](https://user-images.githubusercontent.com/66452376/153454271-b146d7d2-a2f0-4e02-9f3e-1e0d2cc161af.png) included within the issue.

- I added custom icons to each panel for control over the size and color of the accordion arrows. Each arrow's size has been increased to 45px for visibility.

- I added `v-if`  `v-else` blocks to conditionally render a check mark icon or the panel number for the panel avatar where necessary

- I implemented a `determineStyles` method and a `isActive` method which render the appropriate styling/behavior depending on the status of the panel (which I've defined as being in 3 possible states: `default`, `inProgress`, or `completed`)

- I configured the next button to mark previous panel as completed automatically - Im thinking we'll probably need some method of validation to check if the process was actually successful before marking a panel as complete and moving on to the next step. Ive hard coded it for now.

https://user-images.githubusercontent.com/65982182/202015057-c5bde126-3ebb-45e9-befa-897e31594eb2.mov

I assumed that once a step is complete we would not want the user to be able to go back and resubmit information, but this can be changed in case we want to allow editing.

--
Thanks for reviewing! Im not very familiar with Vue, so any comments on how I can improve the implementation are greatly appreciated (:

Resolves #45

### CODE REVIEW

**Comment by paramsiddharth** [comment]
@tdo95 Good job, Tee! :) It looks wonderful, definitely more indicative of the form-filling progress than before.
- About the validation, in a different PR once this is merged, can you add a generic (empty, for now) validation method and implement validation logic for the same for each of the 3 stages? It might make for a good new issue.
- I think we definitely wouldn't want to restrict the user from being able to go back to the previous step, so I think you should enable the ability to return back to an earlier step.

Let me know what you think. :) Excellent work!

**Comment by tdo95** [comment]
> About the validation, in a different PR once this is merged, can you add a generic (empty, for now) validation method and implement validation logic for the same for each of the 3 stages? It might make for a good new issue.

Sure, are you saying I should include an empty method now? Yes, I can work on this issue later (:

> I think we definitely wouldn't want to restrict the user from being able to go back to the previous step, so I think you should enable the ability to return back to an earlier step.

Ok, I will fix this @paramsiddharth

**Comment by paramsiddharth** [comment]
> > About the validation, in a different PR once this is merged, can you add a generic (empty, for now) validation method and implement validation logic for the same for each of the 3 stages? It might make for a good new issue.
> 
> Sure, are you saying I should include an empty method now? Yes, I can work on this issue later (:

No no, you don't have to. :) I was suggesting we can do that in a fresh PR for a new issue.

**Comment by tdo95** [comment]
Updated it so that users can return to previous steps @paramsiddharth 

https://user-images.githubusercontent.com/65982182/202559759-3e23ad4d-99a8-4a27-865b-200af83328ce.mov

**Comment by paramsiddharth** [comment]
> Updated it so that users can return to previous steps @paramsiddharth
> 
>  panel-commands-update.mov

@tdo95 Excellent work, Tee. It looks great to me! :)

**Review by paramsiddharth — DISMISSED** [review]
@manishapriya94 Can you have a look at the video attached in the previous comment and confirm if that is the desired behaviour, before we merge the PR? :)

File: src/components/TakeAction.vue [hunk 1]
```diff
@@ -113,9 +167,36 @@ export default {
...
         nextPage () {
+            const previousPanel = this.panel
+            const nextPanel = this.panel + 1
+
+            //update previous panel's status
+            this.panelStatus[previousPanel] = 'completed'
+            //update next panel's status
+            this.panelStatus[nextPanel] = 'inProgress'
+            //move to next panel
             this.panel = this.panel += 1
```
Comment (thyeggman) [inline src/components/TakeAction.vue:178]: Not part of your change, but we should change this so it's less confusing anyways :)

```suggestion
            this.panel += 1
```

**Review by thyeggman — DISMISSED** [review]
One small nit, looks good to me 😄

**Comment by tdo95** [comment]
@thyeggman I agree, thank you for catching that (: I updated with your change.

**Comment by manishapriya94** [comment]
@tdo95 would you like support with the conflict here?

**Comment by tdo95** [comment]
Hi @manishapriya94, just took a look at it, and was able to resolve it (:

**Pull request merged**